usaspending_api/common/recipient_lookups.py

```diff
@@ -5,24 +5,40 @@ def obtain_recipient_uri(
         recipient_name,
         recipient_unique_id,
         parent_recipient_unique_id=None,
-        child_recipient_unique_id=None):
-    if not recipient_unique_id:
-        return create_recipient_uri_without_duns(recipient_name, recipient_unique_id, parent_recipient_unique_id)
-    return fetch_recipient_uri_with_duns(recipient_unique_id, parent_recipient_unique_id, child_recipient_unique_id)
-
-
-def create_recipient_uri_without_duns(recipient_name, recipient_unique_id, parent_recipient_unique_id=None):
+        is_parent_recipient=False):
+    # No need to create a hash when the parent recipient does not exist
+    if is_parent_recipient and not (recipient_name and recipient_unique_id):
```

Comment:
The parent-child definition requires a parent DUNS but not a name. As reference: we have a few thousand parent recipients with no associated name (aka DUNS-only).

This should be sufficient:
```if is_parent_recipient and not recipient_unique_id```